internal/command/quota_model_test.go

```diff
@@ -0,0 +1,300 @@
+	}, {
+		name: "change from to zero value",
+		fields: fields{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: make([]*quota.SetEventNotification, 0),
+		},
+		args: args{
+			amount:        5,
+			from:          time.Time{},
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: make([]*QuotaNotification, 0),
+		},
+		wantEventPayloadJSON: `{"unit":0,"from":"0001-01-01T00:00:00Z"}`,
+		wantErr:              assert.NoError,
+	}, {
+		name: "add notification",
+		fields: fields{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*quota.SetEventNotification{{
+				ID:      "notification1",
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+		},
+		args: args{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*QuotaNotification{{
+				Percent: 20,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+			idGenerator: id_mock.NewIDGeneratorExpectIDs(t, "notification1"),
+		},
+		wantEventPayloadJSON: `{"unit":0,"notifications":[{"id":"notification1","percent":20,"repeat":true,"callUrl":"https://call.url"}]}`,
+		wantErr:              assert.NoError,
+	}, {
+		name: "change nothing with notification",
+		fields: fields{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*quota.SetEventNotification{{
+				ID:      "notification1",
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+		},
+		args: args{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*QuotaNotification{{
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+			idGenerator: id_mock.NewIDGenerator(t),
+		},
+		wantEventPayloadJSON: `{"unit":0}`,
+		wantErr:              assert.NoError,
+	}, {
+		name: "change notification to zero value",
+		fields: fields{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*quota.SetEventNotification{{
+				ID:      "notification1",
+				Percent: 10,
+				Repeat:  true,
+				CallURL: "https://call.url",
+			}},
+		},
+		args: args{
+			amount:        5,
+			from:          time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			resetInterval: time.Hour,
+			limit:         true,
+			notifications: []*QuotaNotification{},
+		},
+		wantEventPayloadJSON: `{"unit":0,"notifications":[]}`,
+		wantErr:              assert.NoError,
+	}, {
+		name: "create new without notification",
```

Comment:
Can you please also add at least a test for that if the order of the notifications changes with additional notifications defined?

Reply:
done https://github.com/zitadel/zitadel/pull/6597/commits/d07f0b060610f515d064ebd807e96195d77dda3f